### DISCUSSION TITLE
Is there any way to run an entire fixture using a dotnet test command ?

### DISCUSSION THREAD

**Discussion by ayankumaar** [nunit/nunit#3865] [General]
I want to run all the tests present inside the fixture "SmokeTest". So, I am running it using the complete namespace followed by the fixture name. Here is the command that I am running,

**dotnet test --filter "FullyQualifiedName~TestScripts.Tests.SmokeTest"**

**TestScripts.Tests is the namespace inside which the fixture SmokeTest is written. But, it runs all the tests from SmokeTest and also all the tests from SmokeTest2 as well.**

The reason why it is picking tests from both SmokeTest and SmokTest2 is "FullyQualifiedName~TestScripts.Tests.SmokeTest" means run all the fixtures that **contains** the name TestScripts.Tests.SmokeTest.

My question is, is there any way to run all the tests only from TestScripts.Tests.SmokeTest ?

**Comment by mikkelbu** [comment]
> My question is, is there any way to run all the tests only from TestScripts.Tests.SmokeTest ?

Not using the filtering language of VSTest (https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test#filter-option-details) as far as I can tell.

You can either specify the filter using NUnit selection language as a runsetting - see https://blog.prokrams.com/2019/12/16/nunit3-filter-dotnet/ - or you can use 
```
dotnet test --filter "FullyQualifiedName~TestScripts.Tests.SmokeTest."
```
(the last `.` should remove the tests from `SmokeTest2`).
